api/proxy/store/cert_manager.go

```diff
@@ -0,0 +1,282 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync/atomic"
+
+	_ "github.com/Layr-Labs/eigenda/api/clients/v2"
+	"github.com/Layr-Labs/eigenda/api/proxy/common"
+	"github.com/Layr-Labs/eigenda/api/proxy/common/types/certs"
+	"github.com/Layr-Labs/eigenda/api/proxy/store/secondary"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+//go:generate mockgen -package mocks --destination ../test/mocks/cert_manager.go . ICertManager
+
+// ICertManager handles EigenDA certificate operations
+type ICertManager interface {
+	// See [CertManager.Put]
+	Put(ctx context.Context, value []byte) ([]byte, error)
+	// See [CertManager.Get]
+	Get(ctx context.Context, versionedCert certs.VersionedCert, opts common.GETOpts) ([]byte, error)
+	// See [CertManager.SetDispersalBackend]
+	SetDispersalBackend(backend common.EigenDABackend)
+	// See [CertManager.GetDispersalBackend]
+	GetDispersalBackend() common.EigenDABackend
+}
+
+// CertManager handles EigenDA certificate operations
+type CertManager struct {
```

Comment:
not a big fan of this name but fail to see better alternatives

Reply:
EigenDAManager?